From 662648e603685911bd927273b2dcc9173598017f Mon Sep 17 00:00:00 2001
From: Nickey Yang <nickey.yang@rock-chips.com>
Date: Thu, 28 Jul 2016 11:06:51 +0800
Subject: [PATCH] ARM: dts: rockchip: add codec es8323 for rk3288 fennec

This patch makes es8323 work well on the RK3288-Fennec boards.

Change-Id: Ia71101363c5cc4a9650c21c5dbebcad4d785ebf8
Signed-off-by: Nickey Yang <nickey.yang@rock-chips.com>
---
 arch/arm/boot/dts/rk3288-fennec.dts | 37 +++++++++++++++++++++++++----
 1 file changed, 33 insertions(+), 4 deletions(-)

diff --git a/arch/arm/boot/dts/rk3288-fennec.dts b/arch/arm/boot/dts/rk3288-fennec.dts
index d222106c9945..f3a226b950c6 100644
--- a/arch/arm/boot/dts/rk3288-fennec.dts
+++ b/arch/arm/boot/dts/rk3288-fennec.dts
@@ -86,11 +86,27 @@
 		simple-audio-card,format = "i2s";
 		simple-audio-card,name = "rockchip,fennec-codec";
 		simple-audio-card,mclk-fs = <512>;
-		simple-audio-card,cpu {
-			sound-dai = <&i2s>;
+
+		simple-audio-card,dai-link@0 {
+			format = "i2s";
+			cpu {
+				sound-dai = <&i2s>;
+			};
+			codec {
+				sound-dai = <&hdmi>;
+			};
+
 		};
-		simple-audio-card,codec {
-			sound-dai = <&hdmi>;
+
+		simple-audio-card,dai-link@1 {
+			format = "i2s";
+			cpu {
+				sound-dai = <&i2s>;
+			};
+
+			codec {
+				sound-dai = <&es8323>;
+			};
 		};
 	};
 
@@ -398,6 +414,19 @@
 	};
 };
 
+&i2c2 {
+	status = "okay";
+
+	es8323: es8323@10 {
+		compatible = "everest,es8323";
+		reg = <0x10>;
+		hp-det-gpio = <&gpio5 19 GPIO_ACTIVE_HIGH>;
+		clock-names = "mclk";
+		clocks = <&cru SCLK_I2S0_OUT>;
+		#sound-dai-cells = <0>;
+	};
+};
+
 &i2c5 {
 	status = "okay";
 };
-- 
2.35.3

